pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/groupby/NoDictionarySingleColumnGroupKeyGenerator.java

```diff
@@ -51,16 +53,20 @@ public class NoDictionarySingleColumnGroupKeyGenerator implements GroupKeyGenera
   private final DataType _storedType;
   private final Map _groupKeyMap;
   private final int _globalGroupIdUpperBound;
+  private final boolean _nullHandlingEnabled;
```

Comment:
Most of the logic are not sharable, so we can consider making a base implementation, then 2 separate class, one for null enabled, one for disabled

Reply:
Added TODO? Will address in next PR since this one became big.